packages/cli/src/utils.ts

```diff
@@ -191,6 +191,28 @@ export function isPnpmManager(): boolean {
   return !!pnpmPath;
 }
 
+// Check if a binary exists and if it's executable
+export function isBinExisted(binName: string): boolean {
+  try {
+    which.sync(binName);
+    return true;
+  } catch (e) {
+    return false;
+  }
+}
+
+// Get the binary path for a package manager
+export function getManagerBinPath(
```

Comment:
I used this way to check: You can do async like this: https://github.com/rehearsal-js/action/blob/master/src/pm.ts (check --version) :)